public/app/plugins/panel/state-timeline/TimelineChart.tsx

```diff
@@ -34,6 +34,19 @@ export class TimelineChart extends React.Component<TimelineProps> {
   static contextType = PanelContextRoot;
   panelContext: PanelContext = {} as PanelContext;
 
+  getValueColor = (frameIdx: number, fieldIdx: number, value: any) => {
+    const field = this.props.frames[frameIdx].fields[fieldIdx];
+
+    if (field.display) {
+      const disp = field.display(value); // will apply color modes
```

Comment:
everytime I see this I cry a bit inside wishing we had a Field class with a default display processor